classes/ThankYous/Api.php

```diff
@@ -0,0 +1,817 @@
+	 * @param int|null        $offset
+	 * @param int[]|null      $thanked_user_ids
+	 * @param array|null      $date_range
+	 * @param int[]|null      $tag_ids
+	 * @return int[]
+	 */
+	public function GetTagsTotalThankYouUses(SecurityContext $context, ?array $orders = null, ?int $limit = null, ?int $offset = null, ?array $thanked_user_ids = null, ?array $date_range = null, ?array $tag_ids = null): array
+	{
+		$extranet_ids = $this->GetVisibleExtranetIds($context);
+
+		return $this->thank_yous_repository->GetTagsTotalThankYouUses($orders, $limit, $offset, $extranet_ids, true, $date_range, $thanked_user_ids, $tag_ids);
+	}
+
+	/**
+	 * Given an array of ThankYous, returns the total likes on each, indexed by their IDs.
+	 *
+	 * @param ThankYou[] $thank_yous
+	 * @return int[]
+	 */
+	public function GetThankYousLikesCount(array $thank_yous): array
+	{
+		$thank_yous_likes = [];
+		foreach ($thank_yous as $thank_you)
+		{
+			if (!($thank_you instanceof ThankYou))
+			{
+				throw new InvalidArgumentException("Failed to Get Thank Yous Likes Count, one or more entities provided is not a Thank You");
+			}
+
+			$id = $thank_you->GetId();
+
+			if (!isset($id))
+			{
+				continue;
+			}
+
+			$thank_yous_likes[$id] = (int) $this->likes_repository->GetCount(ThankYousRepository::AGGREGATION_ID, $id);
+		}
+
+		return $thank_yous_likes;
+	}
+
+	/**
+	 * @param SecurityContext $context
+	 * @param int[]|null      $user_ids
+	 * @param array|null      $date_range
+	 * @param int[]|null      $tag_ids
+	 * @return int
+	 */
+	public function GetTotalUsers(SecurityContext $context, ?array $user_ids = null, ?array $date_range = null, ?array $tag_ids = null): int
+	{
+		$extranet_ids = $this->GetVisibleExtranetIds($context);
+
+		return $this->thank_yous_repository->GetTotalUsers($user_ids, $date_range, $tag_ids, $extranet_ids);
+	}
+
+	public function GetTotalTags(SecurityContext $context, ?array $date_range = null, ?array $thanked_user_ids = null, ?array $tag_ids = null): int
+	{
+		$extranet_ids = $this->GetVisibleExtranetIds($context);
+
+		return $this->thank_yous_repository->GetTotalTags($extranet_ids, true, $date_range, $thanked_user_ids, $tag_ids);
+	}
+
+	/**
+	 * @param int|int[] $object_types_id
+	 * @return string|string[]
+	 * @throws OwnerClassNameException - If the Name of the oClass could not be determined.
+	 */
+	public function GetOwnerClassNames(array $object_types_id)
+	{
+		return $this->utility->GetOwnerClassNames($object_types_id);
+	}
+
+	/**
+	 * Returns an array of Users indexed by their IDs.
+	 *
+	 * @param int[] $user_ids
+	 * @return User[]
+	 */
+	public function GetUsers(array $user_ids)
+	{
+		return $this->thank_yous_repository->GetUsers($user_ids);
+	}
+
+	/**
+	 * Given an array of arrays with offset id = int, oclass = int, returns a distinct list of User IDs.
+	 *
+	 * @param array $owner_classes
+	 * @return int[]
+	 */
+	public function GetDistinctUserIdsFromOwnerClasses(array $owner_classes)
```

Comment:
I think this should be renamed to something simpler: `GetIndividualIds` or `GetOwnerUserIds`. These aren't just owner classes that are being passed in, they are the owners themselves, represented by an owner class and an owner ID.

Also, I noticed you used the word offset a lot when referring to arrays, but it's actually an index. :smile:

Reply:
Fixed in bb8e0a2f70922c37bbd48049eeeddd96ce6ff371